src/halmos/sevm.py

```diff
@@ -463,18 +472,43 @@ def swap(self, n: int) -> None:
 
         self.stack[-(n + 1)], self.stack[-1] = self.stack[-1], self.stack[-(n + 1)]
 
-    def mloc(self, subst: dict = None) -> int:
+    def mloc(self, subst: dict = None, check_size: bool = True) -> int:
         loc: int = int_of(self.pop(), "symbolic memory offset", subst)
-        if loc > MAX_MEMORY_SIZE:
-            raise OutOfGasError(f"MLOAD {loc} > MAX_MEMORY_SIZE")
+        if check_size and loc > MAX_MEMORY_SIZE:
+            raise OutOfGasError(f"memory {loc=} > MAX_MEMORY_SIZE")
         return loc
 
+    def mslice(self, loc: int, size: int) -> ByteVec:
+        """Wraps a memory slice read with a size check."""
+        if not size:
+            return ByteVec()
+
+        if loc + size > MAX_MEMORY_SIZE:
+            raise OutOfGasError(
+                f"memory access with {loc=} {size=} exceeds MAX_MEMORY_SIZE"
+            )
+
+        return self.memory.slice(start=loc, stop=loc + size)
```

Comment:
you may want to factor out `loc + size` to avoid computing them twice.

Reply:
in 778fbde